qtacrylicmainwindow.cpp

```diff
@@ -0,0 +1,196 @@
+{
+    if (m_acrylicHelper.getNoiseOpacity() != value) {
+        m_acrylicHelper.setNoiseOpacity(value);
+        m_acrylicHelper.updateAcrylicBrush(tintColor());
+        update();
+        Q_EMIT noiseOpacityChanged();
+    }
+}
+
+bool QtAcrylicMainWindow::frameVisible() const
+{
+    return m_frameVisible;
+}
+
+void QtAcrylicMainWindow::setFrameVisible(const bool value)
+{
+    if (m_frameVisible != value) {
+        m_frameVisible = value;
+        update();
+        Q_EMIT frameVisibleChanged();
+    }
+}
+
+QColor QtAcrylicMainWindow::frameColor() const
+{
+    return m_acrylicHelper.getFrameColor();
+}
+
+void QtAcrylicMainWindow::setFrameColor(const QColor &value)
+{
+    if (m_acrylicHelper.getFrameColor() != value) {
+        m_acrylicHelper.setFrameColor(value);
+        update();
+        Q_EMIT frameColorChanged();
+    }
+}
+
+qreal QtAcrylicMainWindow::frameThickness() const
+{
+    return m_acrylicHelper.getFrameThickness();
+}
+
+void QtAcrylicMainWindow::setFrameThickness(const qreal value)
+{
+    if (m_acrylicHelper.getFrameThickness() != value) {
+        m_acrylicHelper.setFrameThickness(value);
+        update();
+        Q_EMIT frameThicknessChanged();
+    }
+}
+
+bool QtAcrylicMainWindow::acrylicEnabled() const
+{
+    return m_acrylicOn;
+}
+
+void QtAcrylicMainWindow::setAcrylicEnabled(const bool value)
+{
+    m_acrylicOn = value;
+    if (m_acrylicOn) {
+        setAutoFillBackground(false);
+        setAttribute(Qt::WA_NoSystemBackground);
+        setAttribute(Qt::WA_OpaquePaintEvent);
+        setBackgroundRole(QPalette::Base);
+    }
+}
+
+void QtAcrylicMainWindow::showEvent(QShowEvent *event)
+{
+    QMainWindow::showEvent(event);
+    if (!acrylicEnabled()) {
+        return;
+    }
+
+    static bool inited = false;
+    if (!inited) {
+        const QWindow *win = windowHandle();
+        FramelessWindowsManager::addWindow(win);
+#if 0
+#ifdef Q_OS_WINDOWS
+        // TODO: let the user choose what he wants.
+        if (Utilities::isWin10OrGreater()) {
+            qputenv(_flh_global::_flh_acrylic_forceEnableOfficialMSWin10AcrylicBlur_flag, "True");
+        }
+#endif
+#endif
+        Utilities::setBlurEffectEnabled(win, true);
+        m_acrylicHelper.install(win);
+        m_acrylicHelper.updateAcrylicBrush(tintColor());
+        connect(&m_acrylicHelper, &QtAcrylicEffectHelper::needsRepaint, this, qOverload<>(&QtAcrylicMainWindow::update));
+        update();
+        inited = true;
+    }
+}
+
+void QtAcrylicMainWindow::paintEvent(QPaintEvent *event)
+{
+    QPainter painter(this);
+    if (acrylicEnabled()) {
+        m_acrylicHelper.paintWindowBackground(&painter, event->region());
+    }
+    if (frameVisible()) {
+        m_acrylicHelper.paintWindowFrame(&painter);
+    }
+    QWidget::paintEvent(event);
+}
+
+void QtAcrylicMainWindow::changeEvent(QEvent *event)
+{
+    if( event->type()==QEvent::WindowStateChange )
+        Q_EMIT windowStateChanged();
```

Comment:
I think we don't need this signal. If you want to re-layout, just do it here.

Reply:
I can mode the setContentMargin() here but the toolbar related stuff has to stay in the example implementation I think